tests/stateful.lua

```diff
@@ -0,0 +1,71 @@
+-- compatibility with Lua 5.1/5.2
+local unpack = rawget(table, "unpack") or unpack
+
+function gen_stateful_iter(values)
+    local i = 1
+    local gen = function(_param, values)
+        if i > #values then
+            return nil
+        end
+        local t = values[i]
+        i = i + 1
+        if type(t) == 'table' then
+            return values, unpack(t)
```

Comment:
`unpack(t)` instead of `unpack(t, 1, table.maxn(t))` causes the test (`Multireturn with nil`) failures when run under LuaJIT:
```sh
luajit ./runtest *.lua
Testing basic.lua
Testing compositions.lua
Testing filtering.lua
Testing generators.lua
Testing indexing.lua
Testing operators.lua
Testing reducing.lua
Testing slicing.lua
Testing stateful.lua
--- stateful.lua        2024-10-09 10:29:36.508803279 +0300
+++ stateful.lua.new    2024-10-09 11:36:29.570003141 +0300
@@ -42,11 +42,11 @@
 
 -- Multireturn with nil
 dump(gen_stateful_iter({{1, nil, 10}, {3, nil, 30}, {5, nil, 50}, {6, nil, 60}, {3, nil, 20}}):drop_while(function(x) return x < 5 end))
 --[[test
-5 nil 50
-6 nil 60
-3 nil 20
+5
+6
+3
 --test]]
 
 -- Multireturn with condition on second returned value
 dump(gen_stateful_iter({{0, 1}, {0, 3}, {0, 5}, {0, 4}}):drop_while(function(x, y) return y < 5 end))
Testing transformations.lua
All tests have passed!
```

So, this part should be adjusted too (as the tested code).

Side note:

`All tests have passed!` line looks implausible...

I've added the following patch:
* `--color` to make output more readable)
* fix for the status check:

```diff
diff --git a/tests/runtest b/tests/runtest
index d1fc4fc..f7c875e 100755
--- a/tests/runtest
+++ b/tests/runtest
@@ -75,9 +75,9 @@ local function process(test_name)
 
     print = prev_print
 
-    local r = os.execute(string.format('diff -U4 "%s" "%s" 2>&1',
+    local r = os.execute(string.format('diff -U4 --color "%s" "%s" 2>&1',
         test_name, new_name))
-    if r then
+    if r == 0 then
         os.remove(new_name)
         return true
     else
```

I suppose these changes should be done inside a separate PR, so just bringing the attention.

Reply:
> unpack(t) instead of unpack(t, 1, table.maxn(t))

Thanks! Fixed the test.

> All tests have passed! line looks implausible...

Yep, it really looks :smile: As you said, this problem is for a separate PR, so I haven't fixed it here.